[WIP] Add support for CRDB target

cjfinnell opened the pull request:
Initial test structure for "just target a CRDB instance and see what breaks" - notable changes that will have to happen:

- [ ] Cannot use `triggers` for tracking changes on CRDB; we could get a little funky with `CHANGEFEED`, or just manually write those changes to the changeset table
- [ ] Cannot use custom `functions` in CRDB - how we verify consistency will have to update for inline queries

cjfinnell committed:
test: Add test harness for CRDB target

cjfinnell commented:
@cornerkitten @svysali tagging you two just for general eyes, but my goal for this PR is to have a running integration test that uses a `cockroachdb/cockroach` single-node instance as the target db.

svysali commented:
`Cannot use triggers for tracking changes on CRDB` -> Warppipe has two modes LR (Logical replication) and Audit. The Logical replication mode does not use triggers - it uses Postgres Logical replication (decode wal messages and applies them to the target). Currently, the warppipe support for this is incomplete. I showed Sam the minimal changes that had to be done to get the sync from source to target done in the LR mode. @cornerkitten do you have the changes in a branch?

cjfinnell committed:
feat: Integrate changes from other WIP PR

Commit in WIP PR: 45e978292c1ebd30c33cfaf8ce2180e56f380723
This differs by adding a rudimentary 'logicalReplicationMode' flag instead
of commenting out behavior.

cjfinnell committed:
chore: Rename int test files for clarity

cjfinnell commented:
Closing as our approach has moved to use different tooling
